app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt

```diff
@@ -307,7 +320,22 @@ class ThreadController @Inject constructor(
         }
 
         fun deleteSearchThreads(realm: MutableRealm) = with(realm) {
-            delete(query<Thread>("${Thread::isFromSearch.name} == true").find())
+            delete(query<Thread>("${Thread::isFromSearch.name} == true"))
+        }
+
+        fun deleteEmptyThreadsInFolder(folderId: String, realm: Realm) {
+            realm.writeBlocking {
+                val emptyThreadsQuery = getEmptyThreadsInFolderQuery(folderId, realm = this)
+                val emptyThreads = emptyThreadsQuery.find()
+                // TODO: Find why we are sometimes displaying empty Threads, and fix it instead of just deleting them.
+                //  It's possibly because we are out of sync, and the situation will resolve by itself shortly?
+                if (emptyThreads.isNotEmpty()) {
+                    emptyThreads.forEach {
+                        SentryDebug.sendEmptyThread(it, "No Message in a Thread when refreshing a Folder")
+                    }
+                    delete(emptyThreadsQuery)
+                }
```

Comment:
Do we want to send a sentry for each thread as in "this thread poses an issue" or do we want to group them all in a single sentry as a "An issue with thread as been detected, here is the list of impacted threads" and save some Sentry events?

Reply:
Yes, for each.